Lib/typing.py

```diff
@@ -1137,7 +1137,11 @@ def __mro_entries__(self, bases):
         res = []
         if self.__origin__ not in bases:
             res.append(self.__origin__)
-        i = bases.index(self)
+        try:
+            i = bases.index(self)
+        except ValueError:
+            # It might not be there, if `Annotated[]` is used:
+            i = 0
```

Comment:
I think this fails in cases like:
```python
class C(typing.Annotated[typing.Callable[[], int], 123]):
    pass
```
`C.__bases__` should include `Generic`. Another similar case that already fails is
```
class C(typing.Generic[T], typing.Annotated[int, 123]):
    pass
```
`C.__bases__` lost `Generic`.

Reply:
`Generic` and `Annotated` together is not tested, please open a new issue about it.

Reply:
@AlexWaygood sorry, off-by-one in the first example. Should be
```python
class B: pass
class C(B, typing.Annotated[typing.Callable[[], int], 123]):
    pass
```
The point is that catching exceptions and setting `i = 0` introduces subtle bugs.

Reply:
> The point is that catching exceptions and setting `i = 0` introduces subtle bugs.

Indeed, good catch. Minimal repro:

```pycon
>>> import types
>>> from typing import *
>>> class A: pass
...
>>> types.new_class('B', (A, Callable[[], int])).__mro__
(<class 'abc.B'>, <class '__main__.A'>, <class 'collections.abc.Callable'>, <class 'typing.Generic'>, <class 'object'>)
>>> types.new_class('B', (A, Annotated[Callable[[], int], 123])).__mro__
(<class 'abc.B'>, <class '__main__.A'>, <class 'collections.abc.Callable'>, <class 'object'>)
```